src/components/Slider/SliderHooks.js

```diff
@@ -0,0 +1,83 @@
+import { useEffect, useRef, useState } from "react";
+
+function ensureSingleValueText(valueText, value, formatter) {
+  if (valueText) {
+    return valueText;
+  }
+  if (typeof value === "undefined") {
+    return undefined;
+  }
+  if (typeof formatter !== "function") {
+    return value;
+  }
+  return formatter(value);
+}
+
+function ensureValueText(valueText, value, formatter) {
+  if (!Array.isArray(value)) {
+    return ensureSingleValueText(valueText, value, formatter);
+  }
+  return value.map((valueSingle, index) => {
+    const valueTextSingle = Array.isArray(valueText) ? valueText[index] : undefined;
+    return ensureSingleValueText(valueTextSingle, valueSingle, formatter);
+  });
+}
+
+export function useIsStateControlledFromOutside(value) {
+  const [isControlled] = useState(typeof value !== "undefined");
```

Comment:
if this can't be changed, why not just use useRef ?

Reply:
Technically it can be changed from consumer - not recommended, and probably not real use case.
I need to think about it.

Reply:
According to our agreements moved to Backlog: Slider Improvements tasks